concordium-consensus/src/Concordium/GlobalState/Account.hs

```diff
@@ -293,17 +324,30 @@ makeAccountHashV2 AccountHashInputsV2{..} = Hash.hashLazy $ runPutLazy $ do
     put ahi2StakedBalance
     put ahi2MerkleHash
 
+-- | Generate the hash for an account (for 'AccountV3'), given the
+--  'AccountHashInputsV2'. 'makeAccountHash' should be used in preference to this function.
+makeAccountHashV3 :: AccountHashInputsV2 av -> Hash.Hash
+makeAccountHashV3 AccountHashInputsV2{..} = Hash.hashLazy $ runPutLazy $ do
+    putShortByteString "AC03"
+    put ahi2NextNonce
+    put ahi2AccountBalance
+    put ahi2StakedBalance
+    put ahi2MerkleHash
+
 -- | Inputs for computing the 'AccountHash' for an account.
 data AccountHashInputs (av :: AccountVersion) where
     AHIV0 :: AccountHashInputsV0 'AccountV0 -> AccountHashInputs 'AccountV0
     AHIV1 :: AccountHashInputsV0 'AccountV1 -> AccountHashInputs 'AccountV1
     AHIV2 :: AccountHashInputsV2 'AccountV2 -> AccountHashInputs 'AccountV2
+    AHIV3 :: AccountHashInputsV2 'AccountV3 -> AccountHashInputs 'AccountV3
```

Comment:
I don't really understand what you mean by this. There are 4 account versions, each with its own `AccountHashInputs` constructor. Do you mean the fact that we reuse `AccountHashInputsV0` and `AccountHashInputsV2` (albeit at different account versions)? This is because the structure of the hashing is the same at that level of abstraction.

Reply:
yes, I expected that every `AccountVersion` has it's own `AccountHashInputs`. Would again be great to have docs making this clear.